src/natcap/invest/urban_nature_access.py

```diff
@@ -2528,6 +2527,40 @@ def _create_mask(*raster_arrays):
         _create_mask, target_mask_path, gdal.GDT_Byte, nodata_target=255)
 
 
+def _warp_lulc(source_lulc_path, target_lulc_path, target_pixel_size,
+               target_bounding_box):
+    """Warp a LULC raster and set a nodata if needed.
+
+    Args:
+        source_lulc_path (str): The path to a source LULC raster.
+        target_lulc_path (str): The path to the new LULC raster.
+        target_pixel_size (tuple): A 2-tuple of the target pixel size.
+        target_bounding_box (tuple): A 4-tuple of the target bounding box.
+
+    Returns:
+        ``None``.
+    """
+    source_raster_info = pygeoprocessing.get_raster_info(source_lulc_path)
+    target_nodata = source_raster_info['nodata'][0]
+    if target_nodata is None:
+        # Guarantee that our nodata cannot be represented by the datatype -
+        # select a nodata value that's out of range.
+        target_nodata = pygeoprocessing.choose_nodata(
+            source_raster_info['numpy_type']) + 1
+
+    pygeoprocessing.warp_raster(
+        source_lulc_path, target_pixel_size, target_lulc_path,
+        'near', target_bb=target_bounding_box,
+        target_projection_wkt=source_raster_info['projection_wkt'])
+
+    # if there is no defined nodata, set a default value
+    raster = gdal.OpenEx(target_lulc_path, gdal.GA_Update)
+    band = raster.GetRasterBand(1)
+    band.SetNoDataValue(target_nodata)
```

Comment:
Forgive me if I'm just misreading something, but could this `SetNoDataValue` unintentionally overwrite an existing nodata value that was defined already?

Reply:
You know, it _shouldn't_ be an issue because `target_nodata` would just be set to the same value it was at before, but this isn't really a clean approach.  I just updated the function in 94f7f006e to only set the nodata value if one isn't already defined.